public/app/features/alerting/unified/components/rule-editor/labels/LabelsField.tsx

```diff
@@ -0,0 +1,477 @@
+import { css, cx } from '@emotion/css';
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { Controller, FormProvider, useFieldArray, useForm, useFormContext } from 'react-hook-form';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { Button, Field, InlineLabel, Input, LoadingPlaceholder, Space, Stack, Text, useStyles2 } from '@grafana/ui';
+import { useDispatch } from 'app/types';
+
+import { labelsApi } from '../../../api/labelsApi';
+import { usePluginBridge } from '../../../hooks/usePluginBridge';
+import { useUnifiedAlertingSelector } from '../../../hooks/useUnifiedAlertingSelector';
+import { fetchRulerRulesIfNotFetchedYet } from '../../../state/actions';
+import { SupportedPlugin } from '../../../types/pluginBridges';
+import { RuleFormValues } from '../../../types/rule-form';
+import AlertLabelDropdown from '../../AlertLabelDropdown';
+import { AlertLabels } from '../../AlertLabels';
+import { NeedHelpInfo } from '../NeedHelpInfo';
+
+import { AddButton, RemoveButton } from './LabelsButtons';
+
+const useGetOpsLabelsKeys = (skip: boolean) => {
+  const { currentData, isLoading: isloadingLabels } = labelsApi.endpoints.getLabels.useQuery(undefined, {
+    skip,
+  });
+  return { loading: isloadingLabels, labelsOpsKeys: currentData };
+};
+const useGetCustomLabels = (dataSourceName: string): { loading: boolean; labelsByKey: Record<string, Set<string>> } => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchRulerRulesIfNotFetchedYet(dataSourceName));
```

Comment:
This is beyond of the scope of this PR but maybe using `alertRuleApi.rulerRules` we could simplify some parts of this code?

Reply:
In the alert rule form we are already fetching ruler rules (in the `FolderAndGroup` component, for example), and we are using the Redux store for it. Moving the logic to a new RTWQ query for this pice of code would imply one additional request that we could skip using the Redux selectors (we are using `fetchRulerRulesIfNotFetchedYet`, so we don't fetch ruler rules if they are already fetched).